views/js/previewer/plugins/tools/scale/scale.js

```diff
@@ -0,0 +1,532 @@
+                height: sizeSettings.height + 275
+            };
+
+            if (requiredSize.width > screenSize.width) {
+                scaleValues.x = screenSize.width / requiredSize.width;
+            }
+
+            if (requiredSize.height > screenSize.height) {
+                scaleValues.y = screenSize.height / requiredSize.height;
+            }
+
+            return Math.min(scaleValues.x, scaleValues.y);
+        },
+
+        /**
+         * Scale devices down to fit screen
+         * @private
+         */
+        scaleFrame: function scale() {
+            var scaleFactor = this.computeScaleFactor();
+
+            var $scaleContainer = controls.$scaleWrapper.find('.preview-scale-container'),
+                _scaleFactor = selectedDevice === 'standard' ? 1 : scaleFactor,
+                containerScaledWidth = $scaleContainer.width() * _scaleFactor,
+                left = (screenSize.width - containerScaledWidth) / 2;
+
+            $scaleContainer.css({
+                left: left > 0 ? left : 0,
+                '-webkit-transform': 'scale(' + _scaleFactor + ',' + _scaleFactor + ')',
+                '-ms-transform': 'scale(' + _scaleFactor + ',' + _scaleFactor + ')',
+                'transform': 'scale(' + _scaleFactor + ',' + _scaleFactor + ')',
+                '-webkit-transform-origin': '0 0',
+                '-ms-transform-origin': '0 0',
+                'transform-origin': '0 0'
+            });
+        },
+
+        /**
+         * position the preview depending on the height of the toolbar
+         * @private
+         */
+        positionPreview: function positionPreview() {
+            var topBarHeight = testRunner.getAreaBroker().getHeaderArea().outerHeight();
+
+            controls.$scaleWrapper.css('top', topBarHeight+ 'px');
+            controls.$scaleWrapper.css('max-height', 'calc(100vh - ' + topBarHeight + 'px )');
+        },
+
+        /**
+         * sets orientation of selected device globally and does some configuration around
+         * @param {String} newOrientation - Device's orientation  could be: 'landscape' or 'portrait'
+         * @private
+         */
+        setOrientation: function setOrientation(newOrientation) {
+            var re;
+            var previewFrame;
+
+            if (newOrientation === orientation) {
+                return;
+            }
+
+            re = new RegExp(orientation, 'g');
+            previewFrame = controls.$scaleWrapper.find('.preview-outer-frame')[0];
+            previewFrame.className = previewFrame.className.replace(re, newOrientation);
+            orientation = newOrientation;
+        },
+
+        /**
+         * Configures screen size configuration state according to passed value from affected control
+         * and calls this.scaleFrame() after all.
+         * @param {Event} event - event object related to affected control.
+         * @private
+         */
+        onDeviceChange: function onDeviceChange(event){
```

Comment:
Is the event really needed here? Only having the target element would be enough.